include/glow/Optimizer/FunctionPass.h

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_OPTIMIZER_FUNCTIONPASS_H
+#define GLOW_OPTIMIZER_FUNCTIONPASS_H
+
+namespace glow {
+
+class FunctionPass {
+protected:
+  /// Perform some tasks before the pass.
+  virtual void doInitialization(Function *F) const {
+    LOG_SCOPE(F->getLogContext(), getName());
+  }
+
+  /// Perform some tasks after the pass.
+  virtual void doFinalization(Function *F) const {}
+
+  /// Implementation of the pass. \returns whether the pass modifies \p F.
+  virtual bool passImpl(Function *F) = 0;
+
+public:
+  FunctionPass() = default;
+  virtual ~FunctionPass() = default;
+
+  // Run the pass, including initialization and finalization. \returns whether
+  // the pass modifies \p F.
+  virtual bool runOnFunction(Function *F) {
```

Comment:
Tiny nit: Do we expect `runOnSomethingElse`? If not, maybe it should be called just `run` as it is obvious from the type of the single argument that it runs on a function.